http/nodepeers.go

```diff
@@ -0,0 +1,55 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/attestantio/go-eth2-client/api"
+	apiv1 "github.com/attestantio/go-eth2-client/api/v1"
+)
+
+type peersResponseJSON struct {
+	Data []apiv1.Peer           `json:"data"`
+	Meta map[string]interface{} `json:"meta,omitempty"`
+}
+
+func (s *Service) NodePeers(ctx context.Context, opts *api.PeerOpts) (*api.Response[*apiv1.Peers], error) {
+	var peerResonse peersResponseJSON
+	// all options are considered optional
+	request := "/eth/v1/node/peers"
+	additionalFields := ""
+
+	for _, stateFilter := range opts.State {
+		additionalFields += fmt.Sprintf("state=%s&", stateFilter)
+	}
+
+	for _, directionFilter := range opts.Direction {
+		additionalFields += fmt.Sprintf("direction=%s&", directionFilter)
+	}
+
+	if len(additionalFields) > 0 {
+		request += "?" + additionalFields
+		// remove the trailing '&'
+		request = strings.TrimSuffix(request, "&")
+	}
```

Comment:
I would make `additionalFields` a `[]string` and add each query parameter to the array individually.  Then you can easily combine them with a `fmt.Sprintf("%s?%s",request,strings.Join(additionalFields,"&"))`

This reduces allocations from repeatedly appending to the string.

Reply:
Ah interesting I reverted that as it felt clunky to make a list, iterate through the opts and then at the end check if the list was nonempty then append the strings. Ill revert to what I had if you prefer.